### ISSUE TITLE
MKV Test file does not play at correct aspect ratio on Fire Stick 4K 

### ISSUE THREAD

**Issue by Pentaphon** [nova-video-player/aos-AVP#259]
Hi, I found a file that plays at the correct aspect ratio on VLC, Kodi, mpv, MPC-HC, and Nova on my phone (Moto G7), but not Nova Video Player on a Fire Stick 4K. When played on the Fire Stick 4K, this test file is stretched to fill up the entire screen instead of being displayed at its original 2.35;1 aspect ratio. Again, this file plays perfectly fine on my phone's Nova Player, but not with Nova on the Fire Stick 4K. 

https://fritschserver.no-ip.org/owncloud/index.php/s/uzmjyEmlc2rGcCT

Thought you should have a look so you can figure out what is going on if you also have this device.

I used this version of Nova sideloaded to Fire Stick 4K (since the Amazon store doesn't have the latest version)
https://github.com/nova-video-player/aos-AVP/releases/tag/v4.22

**Comment by courville** [comment]
Thanks for reporting. Could you please try out his apk that has a new aspect ratio logic and report?
https://drive.google.com/open?id=129SxewI4jvtyy4b7N6-uJwl0YQfeGe1m

**Comment by courville** [comment]
OK I found a firestick4k and aspect ratio is corrected with the proposed fix.
What happens is that the codec reports 1920x1088 resolution and real video as reported is 1280x544. Applying what proposes the codec results in stretched fullscreen video.
It is worth noting that normally codec should report correct resolution and this is enforced on AndroidTV by proper CTS tests. But hey... amazon is almighty.
```
12-13 23:35:04.457   275  1056 D MtkOmxVdecEx: [0xa818d000] MtkOmxVdec::FreeBuffer all output buffers have been freed!!! signal mOutPortFreeDoneSem(1)
12-13 23:35:04.458   275  1056 D MtkOmxMVAMgr: [0xa851b5c8] Remove ion MVA Map, remain buffer count=0
12-13 23:35:04.459   275  1056 D MtkOmxVdecEx: [0xa818d000] MtkOmxVdec::SendCommand cmd=OMX_CommandPortEnable
12-13 23:35:04.459   275  6099 D MtkOmxVdecEx: [0xa818d000] # Got general command (OMX_CommandPortEnable)
12-13 23:35:04.459   275  6099 D MtkOmxVdecEx: [0xa818d000] Wait on mOutPortAllocDoneSem(0)
12-13 23:35:04.459   275  1056 D MtkOmxVdecEx: [0xa818d000] [MtkOmxVdec] Get decoder property, VDEC_DRV_QUERY_TYPE_UFO_SUPPORT: 1, 4489
12-13 23:35:04.459   275  1056 D MtkOmxVdecEx: [0xa818d000] mOutputPortDef eColorFormat(7f000001), eColorFormat(7f000001), meDecodeType(0), mForceOutputBufferCount(0), mIsUsingNativeBuffers(1)
12-13 23:35:04.459   275  1056 D MtkOmxVdecEx: [0xa818d000] [MtkOmxVdec] Get decoder property, VDEC_DRV_QUERY_TYPE_UFO_SUPPORT: 1, 4489
12-13 23:35:04.459   275  1056 D MtkOmxVdecEx: [0xa818d000] 32x32 Aligned! mOutputPortDef.nBufferSize(3133440), nStride(1920), nSliceHeight(1088) nBufferCountActual(13)
12-13 23:35:04.459   275  1056 D MtkOmxVdecEx: [0xa818d000] [MtkOmxVdec] Get decoder property, VDEC_DRV_QUERY_TYPE_UFO_SUPPORT: 1, 4489
12-13 23:35:04.459   275  1056 D MtkOmxVdecEx: [0xa818d000] mOutputPortDef eColorFormat(7f000001), eColorFormat(7f000001), meDecodeType(0), mForceOutputBufferCount(0), mIsUsingNativeBuffers(1)
12-13 23:35:04.459   275  1056 D MtkOmxVdecEx: [0xa818d000] [MtkOmxVdec] Get decoder property, VDEC_DRV_QUERY_TYPE_UFO_SUPPORT: 1, 4489
12-13 23:35:04.459   275  1056 D MtkOmxVdecEx: [0xa818d000] 32x32 Aligned! mOutputPortDef.nBufferSize(3133440), nStride(1920), nSliceHeight(1088) nBufferCountActual(13)
12-13 23:35:04.460  5974  6097 D SurfaceUtils: set up nativeWindow 0x8c26c808 for 1920x1088, color 0x7f000001, rotation 0, usage 0x2933
12-13 23:35:04.460   275  1056 E OMXNodeInstance: setParameter(1130047:MTK.DECODER.AVC, ParamPortDefinition(0x2000001)) ERROR: BadParameter(0x80001005)
12-13 23:35:04.460  5974  6097 W ACodec  : [OMX.MTK.VIDEO.DECODER.AVC] setting nBufferCountActual to 14 failed: -2147483648
12-13 23:35:04.461   275  1056 D MtkOmxVdecEx: [0xa818d000] UseBuffer:: clear mFrameBuf
12-13 23:35:04.461   275  1056 D MtkOmxMVAMgr: [0xa851b5c8] [WARNING] Can not find MVA from VA: 0xa82fc540
12-13 23:35:04.463   275  1056 D MtkOmxVdecEx: [0xa818d000] signal mOutPortAllocDoneSem (1)
12-13 23:35:04.463   275  1056 D MtkOmxVdecEx: [0xa818d000] output port populated
12-13 23:35:04.464   275  6099 D MtkOmxVdecEx: [0xa818d000] mPortReconfigInProgress as FALSE
12-13 23:35:04.495  5974  6096 I MediaCodecLogger: updateFormatChanged width = 1280 height = 544
12-13 23:35:04.499  5974  6096 I MediaCodecLogger: updatePTSTime [HW.video.avc] First PTS after Flush or reset = 14556000
12-13 23:35:04.499  5974  6096 I MediaCodecLogger: [ASAP] HW.video.avc Got First Frame Ready 434448
12-13 23:35:04.501  5974  6096 I MediaCodecLogger: [ASAP] HW.video.avc Got First Frame Render 434450
12-13 23:35:04.501  5974  6096 I MediaCodecLogger: App uses Timestamps for AVSync
12-13 23:35:04.502  5974  5995 D avos_player: init_renderer: width NOT changed: 1280 -> 1920
12-13 23:35:04.502  5974  5995 D avos_player: init_renderer: height NOT changed: 544 -> 1088
12-13 23:35:04.502  5974  5995 D avos_player: sfdec_read: INFO_FORMAT_CHANGED: 1280x544
```

**Comment by courville** [comment]
Full logs attached
[firestick4k.log.zip](https://github.com/nova-video-player/aos-AVP/files/3962991/firestick4k.log.zip)

**Comment by Pentaphon** [comment]
@courville I can confirm you fixed it with the 4.24 build using my Fire Stick 4K. Thanks! Closing!
